app/controllers/subscription_controller.rb

```diff
@@ -46,32 +46,23 @@ def add
           begin
             tag.save!
           rescue ActiveRecord::RecordInvalid
-            flash[:error] = tag.errors.full_messages
-            redirect_to "/subscriptions" + "?_=" + Time.now.to_i.to_s
+            render :json => { status: "500", message: "error" }
```

Comment:
Hi, this is great but I think we need to preserve the original non-JS version too, as this method is used in other places. Could we do that?

Reply:
Yes, we can do that, just need to check if request.xhr return true right?